rust/kernel/types.rs

```diff
@@ -387,3 +395,141 @@ pub enum Either<L, R> {
     /// Constructs an instance of [`Either`] containing a value of type `R`.
     Right(R),
 }
+
+/// A type that can be represented in little-endian bytes.
+pub trait LittleEndian {
+    /// Converts from native to little-endian encoding.
+    fn to_le(self) -> Self;
+
+    /// Converts from little-endian to native encoding.
+    fn to_native(self) -> Self;
```

Comment:
Looks like in C side, it's usually called `_cpu` or `_host` for native byteordering, see `include/byteorder/generic.h